dask/blockwise.py

```diff
@@ -689,17 +789,34 @@ def make_blockwise_graph(func, output, out_indices, *arrind_pairs, **kwargs):
                 args.append(tups)
         out_key = (output,) + out_coords
 
-        if kwargs:
-            val = (apply, func, args, kwargs2)
+        if deserializing:
+            deps.update(func_future_args)
+            args = stringify_collection_keys(args) + list(func_future_args)
+            if kwargs:
+                val = {
+                    "function": dumps_function(apply),
+                    "args": warn_dumps(args),
+                    "kwargs": warn_dumps(kwargs2),
+                }
+            else:
+                val = {"function": func, "args": warn_dumps(args)}
```

Comment:
It would be nice if we could separate out this serialization-specific logic into `__dask_distributed_unpack__` somehow. However this may be difficult and more trouble than it's worth, so we shouldn't let this get in the way of merging. Just wanted to point this out as possible follow up work